tests/Versioning/VersionManagerTest.php

```diff
@@ -420,6 +449,54 @@ public function testRestoreWithNewChildren()
         $this->assertFalse($node->hasNode('newChildNode'));
     }
 
+    public function testRestoreWithChildrenInitialize()
+    {
+        $nodeTypeManager = $this->session->getWorkspace()->getNodeTypeManager();
+
+        $wildcardPropertyDefinition = $nodeTypeManager->createPropertyDefinitionTemplate();
+        $wildcardPropertyDefinition->setName('*');
+
+        $testNodeType = $nodeTypeManager->createNodeTypeTemplate();
+        $testNodeType->setName('nt:test-initialize-children');
+        $testNodeType->getPropertyDefinitionTemplates()->append($wildcardPropertyDefinition);
+
+        $testChildNodeType = $nodeTypeManager->createNodeTypeTemplate();
+        $testChildNodeType->setName('nt:child');
+        $testChildNodeType->getPropertyDefinitionTemplates()->append($wildcardPropertyDefinition);
+
+        $childNodeDefinition = $nodeTypeManager->createNodeDefinitionTemplate();
+        $childNodeDefinition->setName('child');
+        $childNodeDefinition->setDefaultPrimaryTypeName('nt:child');
+        $childNodeDefinition->setRequiredPrimaryTypeNames(['nt:child']);
+        $childNodeDefinition->setAutoCreated(true);
+        $childNodeDefinition->setMandatory(true);
+        $childNodeDefinition->setOnParentVersion(OnParentVersionAction::INITIALIZE);
+        $testNodeType->getNodeDefinitionTemplates()->append($childNodeDefinition);
+
+        $nodeTypeManager->registerNodeType($testChildNodeType, true);
+        $nodeTypeManager->registerNodeType($testNodeType, true);
```

Comment:
@dbu No need to create a jackrabbit PR yet, this is not working... I always get the following exception:

```
PHPCR\RepositoryException: Unexpected error:
CURL Error:
Response (HTTP 400): PROPPATCH
--uri: --
array (
  0 => 'http://localhost:8080/server/testsWorkspace',
)
--curl getinfo: --
array (
  'url' => 'http://localhost:8080/server/testsWorkspace',
  'content_type' => NULL,
  'http_code' => 400,
  'header_size' => 69,
  'request_size' => 699,
  'filetime' => -1,
  'ssl_verify_result' => 0,
  'redirect_count' => 0,
  'total_time' => 0.0012719999999999999,
  'namelookup_time' => 1.7E-5,
  'connect_time' => 1.7E-5,
  'pretransfer_time' => 4.1E-5,
  'size_upload' => 479.0,
  'size_download' => 0.0,
  'speed_download' => 0.0,
  'speed_upload' => 376572.0,
  'download_content_length' => 0.0,
  'upload_content_length' => 479.0,
  'starttransfer_time' => 0.001263,
  'redirect_time' => 0.0,
  'redirect_url' => '',
  'primary_ip' => '::1',
  'certinfo' =>
  array (
  ),
  'primary_port' => 8080,
  'local_ip' => '::1',
  'local_port' => 61399,
)
--request body (size: 479 bytes): --
<?xml version="1.0" encoding="UTF-8" standalone="no"?><D:propertyupdate xmlns:D="DAV:"><D:set><D:prop><dcr:nodetypes-cnd xmlns:dcr="http://www.day.com/jcr/webdav/1.0"><dcr:cnd>&lt;nt='http://www.jcp.org/jcr/nt/1.0'&gt;
[nt:test-initialize-children] &gt; nt:base
mixin query
- * (String)
nofulltext noqueryorder
+ child (nt:child)
= nt:child
mandatory autocreated INITIALIZE
</dcr:cnd><dcr:allowupdate>true</dcr:allowupdate></dcr:nodetypes-cnd></D:prop></D:set></D:propertyupdate>
--response body (size: 0 bytes): --

--end response body--
```

Any idea what I could do about it?